### PR TITLE
Git issues 20210824

### LINKED ISSUE #28: Dates
Change start_year to start_date and change "timesteps" to end_date

n_timesteps can be calculated internally

any objections? Can you take this one on @KeeganShaw-GIS 

@tbarchyn @AkaMrmuffin @KeeganShaw-GIS @c-vollrath

### LINKED ISSUE #45: Axe spin up and count file
I think there's a way to axe spin up and count file and just get better results all around. I'll look into this and report back.

### LINKED ISSUE #64: emission rate parameter names
I think it would add clarity to move from the following for the sake of clarity. Just slightly better (maybe) parameter names - no functional change

**current inputs and comments for leaks**:
```
'leak_file': 'leak_rates.csv',
'subtype_distributions': [False, 'subtype_distributions.csv'], # If True, will overwrite global Leak Dist
'use_empirical_rates': False, # 'sample', 'fit', or False
'leak_rate_dist': ['lognorm',-2.776, 1.462, "kilogram", "hour"]
```

**proposed inputs and comments for leaks**:
```
'leak_dist_source': 'manual', # 'sample', 'fit', or 'manual'
'leak_dist_empirical': 'leak_rates.csv', # Used if leak_dist_source == 'sample' or 'fit'
'leak_dist_manual': ['lognorm',-2.776, 1.462, "kilogram", "hour"] # Used if leak_dist_source == 'manual'
'leak_dist_subtypes': [False, 'subtype_distributions.csv'], # If True, overwrites 'leak_dist_source'
```

@KeeganShaw-GIS  if you agree can you implement and close?

### PR DESCRIPTION
**Pull request by KeeganShaw-GIS**
Standardizes Leaks required for #19
Closes #64
Closes #28 
Closes #45 

_Temporarily resolved -_ The changes  will store and load pre generated leaks / sites. One issue with this that should be resolved before accepting is that  there are no checks to see if program has changed before loading the saved pregenerated values.

### CODE REVIEW

**Comment by KeeganShaw-GIS** [comment]
Update on temporary solution for uploading saved pregenerated values. Now the User is prompted if pregen_leaks is set to true and pregenerated leaks / sites files exist. The user can either use existing pregen data or regenerate. File check or comparison to input parameters should be done, has been moved to issue #108.

File: LDAR_Sim/model_code/default_parameters/default_global_parameters.py [hunk 1]
```diff
@@ -25,12 +25,16 @@
     'input_directory': './inputs_template',
     'output_directory': './outputs',
     'n_processes':  None,
-    'n_simulations': 2,
     'print_from_simulations': True,
+    'n_simulations': 2,
     'spin_up': 0,
```
Comment (tarcadius) [inline LDAR_Sim/model_code/default_parameters/default_global_parameters.py:30]: Is spin_up removed here or in a different PR?

File: LDAR_Sim/model_code/geography/vector.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+def grid_contains_point(point_coord, grid_list_coord):
```
Comment (tarcadius) [inline LDAR_Sim/model_code/geography/vector.py:1]: Can you please add typical header for this with all the jazz, including description of purpose?

File: LDAR_Sim/model_code/initialization/leaks.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+from datetime import datetime, timedelta
```
Comment (tarcadius) [inline LDAR_Sim/model_code/initialization/leaks.py:1]: header + details please :) This is awesome btw.

File: LDAR_Sim/model_code/initialization/sites.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+
```
Comment (tarcadius) [inline LDAR_Sim/model_code/initialization/sites.py:1]: Header please ... and awesome work!

File: LDAR_Sim/model_code/ldar_sim.py [hunk 5]
```diff
@@ -518,8 +445,4 @@ def finalize(self):
             'sites': site_df,
         }
 
-        # Return to original input directory
-        os.chdir(params['input_directory'])
-        os.chdir('..')
-
         return(sim_summary)
```
Comment (tarcadius) [inline LDAR_Sim/model_code/ldar_sim.py:448]: Well this file is greatly cleaned up, good job!

File: LDAR_Sim/model_code/utils/distributions.py [hunk 6]
```diff
@@ -83,3 +80,61 @@ def leak_rvs(distribution, max_size=None, gpsec_conversion=None):
         if not max_size or leaksize < max_size:
             break  # Rerun if value is larger than maximum
     return leaksize
+
+
+def unpackage_dist(program, in_dir=None):
```
Comment (tarcadius) [inline LDAR_Sim/model_code/utils/distributions.py:85]: maybe a description of this function?

**Review by tarcadius — COMMENTED** [review]
Amazing work Keegan! Didn't see any updates to documentation.. did I miss or not needed/

**Review by AkaMrmuffin — APPROVED** [review]
Great job Keegan. Thanks for simplifying/cleaning functions in the mobile_crew.py.

File: LDAR_Sim/model_code/default_parameters/default_global_parameters.py (hunk 1 above)
Comment (KeeganShaw-GIS) [reply LDAR_Sim/model_code/default_parameters/default_global_parameters.py:30]: included in this now

File: LDAR_Sim/model_code/geography/vector.py (hunk 2 above)
Comment (KeeganShaw-GIS) [reply LDAR_Sim/model_code/geography/vector.py:1]: Added

File: LDAR_Sim/model_code/initialization/leaks.py (hunk 3 above)
Comment (KeeganShaw-GIS) [reply LDAR_Sim/model_code/initialization/leaks.py:1]: Added

File: LDAR_Sim/model_code/initialization/sites.py (hunk 4 above)
Comment (KeeganShaw-GIS) [reply LDAR_Sim/model_code/initialization/sites.py:1]: Added

File: LDAR_Sim/model_code/utils/distributions.py (hunk 6 above)
Comment (KeeganShaw-GIS) [reply LDAR_Sim/model_code/utils/distributions.py:85]: updated

**Review by tarcadius — APPROVED** [review]
Looks great, thanks Keegan!

**Pull request merged**